test/mock_osystem.cpp

```diff
@@ -0,0 +1,132 @@
+#define FORBIDDEN_SYMBOL_ALLOW_ALL
+
+#include "mock_osystem.h"
+#include "backends/fs/posix/posix-fs-factory.h"
+
+namespace Common {
+
+class OSystem_Mock : public OSystem {
+public:
+	OSystem_Mock() {
+		_fsFactory = new POSIXFilesystemFactory();
+	}
+	void logMessage(LogMessageType::Type type, const char *message) {
+		fprintf(stderr, "logMessage: %s", message);
+	}
+	Graphics::PixelFormat getScreenFormat() const {
+		assert(0);
+	}
+	Common::List<Graphics::PixelFormat> getSupportedFormats() const {
+		assert(0);
+	}
+	void initSize(uint width, uint height, const Graphics::PixelFormat *format = nullptr) {
+		assert(0);
+	}
+	int16 getHeight() {
+		assert(0);
+	}
+	int16 getWidth() {
+		assert(0);
+	}
+	PaletteManager *getPaletteManager() {
+		assert(0);
+	}
+	void copyRectToScreen(const void *buf, int pitch, int x, int y, int w, int h) {
+		assert(0);
+	}
+	Graphics::Surface *lockScreen() {
+		assert(0);
+	}
+	void unlockScreen() {
+		assert(0);
+	}
+	void fillScreen(uint32 col) {
+		assert(0);
+	}
+	void updateScreen() {
+		assert(0);
+	}
+	void setShakePos(int shakeXOffset, int shakeYOffset) {
+		assert(0);
+	}
+	void showOverlay() {
+		assert(0);
+	}
+	void hideOverlay() {
+		assert(0);
+	}
+	bool isOverlayVisible() const {
+		assert(0);
+	}
+	Graphics::PixelFormat getOverlayFormat() const {
+		assert(0);
+	}
+       	void clearOverlay() {
+		assert(0);
+	}
+	void grabOverlay(void *buf, int pitch) {
+		assert(0);
+	}
+	void copyRectToOverlay(const void *buf, int pitch, int x, int y, int w, int h) {
+		assert(0);
+	}
+	int16 getOverlayHeight() {
+		assert(0);
+	}
+	int16 getOverlayWidth() {
+		assert(0);
+	}
+	bool showMouse(bool visible) {
+		assert(0);
+	}
+	void warpMouse(int x, int y) {
+		assert(0);
+	}
+	void setMouseCursor(const void *buf, uint w, uint h, int hotspotX, int hotspotY, uint32 keycolor, bool dontScale = false, const Graphics::PixelFormat *format = nullptr) {
+		assert(0);
+	}
+	uint32 getMillis(bool skipRecord = false) {
+		assert(0);
+	}
+	void delayMillis(uint msecs) {
+		assert(0);
+	}
+	void getTimeAndDate(TimeDate &t) const {
+		assert(0);
+	}
+	MutexRef createMutex() {
+		assert(0);
+	}
+	void lockMutex(MutexRef mutex) {
+		assert(0);
+	}
+	void unlockMutex(MutexRef mutex) {
+		assert(0);
+	}
+	void deleteMutex(MutexRef mutex) {
+		assert(0);
+	}
+	Audio::Mixer *getMixer() {
+		assert(0);
+	}
+	void quit() {
+		assert(0);
+	}
+	void displayMessageOnOSD(const Common::U32String &msg) {
+		assert(0);
+	}
+	void displayActivityIconOnOSD(const Graphics::Surface *icon) {
+		assert(0);
+	}
+
+	void addSysArchivesToSearchSet(Common::SearchSet &s, int priority) override {
+		Common::String dataPath = "dists/engine-data";
+		Common::FSNode dataNode(dataPath);
+		s.add(dataPath, new Common::FSDirectory(dataNode, 4), priority);
+	}
+};
+
+void install_mock_g_system() {
+	g_system = new OSystem_Mock();
```

Comment:
Would using the `OSystem_NULL` backend not be simpler? Or does it pull too many dependencies?

Reply:
Done

Reply:
done